App is crashing on opening.

**Describe the bug**
As soon as the app is opened, it crashes.

**To Reproduce**
Install the latest version of the app. Open it. The app crashes.

**Expected behavior**
App should not crash.

**Logs**
`2019-05-11 00:53:57.085 31967-31967/com.nilhcem.blenamebadge E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.nilhcem.blenamebadge, PID: 31967
    java.lang.RuntimeException: Cannot create an instance of class com.nilhcem.blenamebadge.ui.fragments.main_text.MainTextDrawableViewModel
        at androidx.lifecycle.ViewModelProvider$NewInstanceFactory.create(ViewModelProvider.java:184)
        at androidx.lifecycle.ViewModelProvider$AndroidViewModelFactory.create(ViewModelProvider.java:241)
        at androidx.lifecycle.ViewModelProvider.get(ViewModelProvider.java:164)
        at androidx.lifecycle.ViewModelProvider.get(ViewModelProvider.java:130)
        at com.nilhcem.blenamebadge.ui.fragments.main_text.MainTextDrawableFragment.inject(MainTextDrawableFragment.kt:71)
        at com.nilhcem.blenamebadge.ui.fragments.main_text.MainTextDrawableFragment.onCreate(MainTextDrawableFragment.kt:67)
        at androidx.fragment.app.Fragment.performCreate(Fragment.java:2528)
        at androidx.fragment.app.FragmentManagerImpl.moveToState(FragmentManagerImpl.java:843)
        at androidx.fragment.app.FragmentTransition.addToFirstInLastOut(FragmentTransition.java:1196)
        at androidx.fragment.app.FragmentTransition.calculateFragments(FragmentTransition.java:1079)
        at androidx.fragment.app.FragmentTransition.startTransitions(FragmentTransition.java:118)
        at androidx.fragment.app.FragmentManagerImpl.executeOpsTogether(FragmentManagerImpl.java:1858)
        at androidx.fragment.app.FragmentManagerImpl.removeRedundantOperationsAndExecute(FragmentManagerImpl.java:1816)
        at androidx.fragment.app.FragmentManagerImpl.execPendingActions(FragmentManagerImpl.java:1723)
        at androidx.fragment.app.FragmentManagerImpl.dispatchStateChange(FragmentManagerImpl.java:2624)
        at androidx.fragment.app.FragmentManagerImpl.dispatchActivityCreated(FragmentManagerImpl.java:2580)
        at androidx.fragment.app.FragmentController.dispatchActivityCreated(FragmentController.java:246)
        at androidx.fragment.app.FragmentActivity.onStart(FragmentActivity.java:541)
        at androidx.appcompat.app.AppCompatActivity.onStart(AppCompatActivity.java:201)
        at android.app.Instrumentation.callActivityOnStart(Instrumentation.java:1392)
        at android.app.Activity.performStart(Activity.java:7157)
        at android.app.ActivityThread.handleStartActivity(ActivityThread.java:2952)
        at android.app.servertransaction.TransactionExecutor.performLifecycleSequence(TransactionExecutor.java:180)
        at android.app.servertransaction.TransactionExecutor.cycleToPath(TransactionExecutor.java:165)
        at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:142)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:70)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1823)
        at android.os.Handler.dispatchMessage(Handler.java:107)
        at android.os.Looper.loop(Looper.java:198)
        at android.app.ActivityThread.main(ActivityThread.java:6729)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
     Caused by: java.lang.InstantiationException: java.lang.Class<com.nilhcem.blenamebadge.ui.fragments.main_text.MainTextDrawableViewModel> has no zero argument constructor
        at java.lang.Class.newInstance(Native Method)
        at androidx.lifecycle.ViewModelProvider$NewInstanceFactory.create(ViewModelProvider.java:182)
        at androidx.lifecycle.ViewModelProvider$AndroidViewModelFactory.create(ViewModelProvider.java:241) 
        at androidx.lifecycle.ViewModelProvider.get(ViewModelProvider.java:164) 
        at androidx.lifecycle.ViewModelProvider.get(ViewModelProvider.java:130) 
        at com.nilhcem.blenamebadge.ui.fragments.main_text.MainTextDrawableFragment.inject(MainTextDrawableFragment.kt:71) 
        at com.nilhcem.blenamebadge.ui.fragments.main_text.MainTextDrawableFragment.onCreate(MainTextDrawableFragment.kt:67) 
        at androidx.fragment.app.Fragment.performCreate(Fragment.java:2528) 
        at androidx.fragment.app.FragmentManagerImpl.moveToState(FragmentManagerImpl.java:843) 
        at androidx.fragment.app.FragmentTransition.addToFirstInLastOut(FragmentTransition.java:1196) 
        at androidx.fragment.app.FragmentTransition.calculateFragments(FragmentTransition.java:1079) 
        at androidx.fragment.app.FragmentTransition.startTransitions(FragmentTransition.java:118) 
        at androidx.fragment.app.FragmentManagerImpl.executeOpsTogether(FragmentManagerImpl.java:1858) 
        at androidx.fragment.app.FragmentManagerImpl.removeRedundantOperationsAndExecute(FragmentManagerImpl.java:1816) 
        at androidx.fragment.app.FragmentManagerImpl.execPendingActions(FragmentManagerImpl.java:1723) 
        at androidx.fragment.app.FragmentManagerImpl.dispatchStateChange(FragmentManagerImpl.java:2624) 
        at androidx.fragment.app.FragmentManagerImpl.dispatchActivityCreated(FragmentManagerImpl.java:2580) 
        at androidx.fragment.app.FragmentController.dispatchActivityCreated(FragmentController.java:246) 
        at androidx.fragment.app.FragmentActivity.onStart(FragmentActivity.java:541) 
        at androidx.appcompat.app.AppCompatActivity.onStart(AppCompatActivity.java:201) 
        at android.app.Instrumentation.callActivityOnStart(Instrumentation.java:1392) 
        at android.app.Activity.performStart(Activity.java:7157) 
        at android.app.ActivityThread.handleStartActivity(ActivityThread.java:2952) 
        at android.app.servertransaction.TransactionExecutor.performLifecycleSequence(TransactionExecutor.java:180) 
        at android.app.servertransaction.TransactionExecutor.cycleToPath(TransactionExecutor.java:165) 
        at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:142) 
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:70) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1823) 
        at android.os.Handler.dispatchMessage(Handler.java:107) 
        at android.os.Looper.loop(Looper.java:198) 
        at android.app.ActivityThread.main(ActivityThread.java:6729) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858) `

**Would you like to work on the issue?**
YES.

--- a/app/src/main/java/com/nilhcem/blenamebadge/ui/drawer/DrawerActivity.kt
+++ b/app/src/main/java/com/nilhcem/blenamebadge/ui/drawer/DrawerActivity.kt
@@ -50,8 +50,7 @@ class DrawerActivity : AppCompatActivity(), NavigationView.OnNavigationItemSelec
     private var viewModel: AppViewModel? = null
 
     private fun inject() {
-        val savedConfigFactory = InjectorUtils.provideFilesViewModelFactory()
-        viewModel = ViewModelProviders.of(this, savedConfigFactory)
+        viewModel = ViewModelProviders.of(this, InjectorUtils.provideFilesViewModelFactory())
             .get(AppViewModel::class.java)
     }
 

--- a/app/src/main/java/com/nilhcem/blenamebadge/ui/fragments/main_saved/MainSavedFragment.kt
+++ b/app/src/main/java/com/nilhcem/blenamebadge/ui/fragments/main_saved/MainSavedFragment.kt
@@ -46,8 +46,7 @@ class MainSavedFragment : BaseFragment() {
     }
 
     override fun inject() {
-        val savedConfigFactory = InjectorUtils.provideFilesViewModelFactory()
-        viewModel = ViewModelProviders.of(this, savedConfigFactory)
+        viewModel = ViewModelProviders.of(this, InjectorUtils.provideFilesViewModelFactory())
             .get(AppViewModel::class.java)
     }
 

--- a/app/src/main/java/com/nilhcem/blenamebadge/ui/fragments/main_text/MainTextDrawableFragment.kt
+++ b/app/src/main/java/com/nilhcem/blenamebadge/ui/fragments/main_text/MainTextDrawableFragment.kt
@@ -37,6 +37,7 @@ import com.nilhcem.blenamebadge.data.util.SendingData
 import com.nilhcem.blenamebadge.ui.custom.knob.Croller
 import com.nilhcem.blenamebadge.ui.fragments.base.BaseFragment
 import com.nilhcem.blenamebadge.util.Converters
+import com.nilhcem.blenamebadge.util.InjectorUtils
 import com.nilhcem.blenamebadge.util.SendingUtils
 import kotlinx.android.synthetic.main.effects_layout.view.*
 import kotlinx.android.synthetic.main.fragment_main_text.view.*
@@ -68,7 +69,9 @@ class MainTextDrawableFragment : BaseFragment() {
     }
 
     override fun inject() {
-        viewModel = ViewModelProviders.of(this).get(MainTextDrawableViewModel::class.java)
+        viewModel = ViewModelProviders
+                .of(this, InjectorUtils.provideMainTextDrawableViewModelFactory())
+                .get(MainTextDrawableViewModel::class.java)
     }
 
     override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {

--- a/app/src/main/java/com/nilhcem/blenamebadge/ui/fragments/main_text/MainTextDrawableViewModelProviderFactory.kt
+++ b/app/src/main/java/com/nilhcem/blenamebadge/ui/fragments/main_text/MainTextDrawableViewModelProviderFactory.kt
@@ -1,0 +1,14 @@
+package com.nilhcem.blenamebadge.ui.fragments.main_text
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import com.nilhcem.blenamebadge.data.fragments.FilesRepository
+
+class MainTextDrawableViewModelProviderFactory(private val configRepository: FilesRepository)
+    : ViewModelProvider.NewInstanceFactory() {
+
+    override fun <T : ViewModel?> create(modelClass: Class<T>): T {
+        @Suppress("UNCHECKED_CAST")
+        return MainTextDrawableViewModel(configRepository) as T
+    }
+}

--- a/app/src/main/java/com/nilhcem/blenamebadge/util/InjectorUtils.kt
+++ b/app/src/main/java/com/nilhcem/blenamebadge/util/InjectorUtils.kt
@@ -3,6 +3,7 @@ package com.nilhcem.blenamebadge.util
 import com.nilhcem.blenamebadge.data.fragments.FilesContainer
 import com.nilhcem.blenamebadge.data.fragments.FilesRepository
 import com.nilhcem.blenamebadge.ui.AppViewModelProviderFactory
+import com.nilhcem.blenamebadge.ui.fragments.main_text.MainTextDrawableViewModelProviderFactory
 
 object InjectorUtils {
 
@@ -10,5 +11,9 @@ object InjectorUtils {
 
     fun provideFilesViewModelFactory(): AppViewModelProviderFactory {
         return AppViewModelProviderFactory(filesRepo)
+    }
+
+    fun provideMainTextDrawableViewModelFactory(): MainTextDrawableViewModelProviderFactory {
+        return MainTextDrawableViewModelProviderFactory(filesRepo)
     }
 }

